conbenchlegacy testing bug: werkzeug import error

In an unrelated PR, `conbenchlegacy` unit tests are failing with a werkzeug import error: https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496

```python
Run pytest -vv legacy
  pytest -vv legacy
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_API_TOKEN: ***
    pythonLocation: /opt/hostedtoolcache/Python/3.[1](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:1)0.13/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.10.13/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.13/x64
    Python[2](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:2)_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:3).10.13/x6[4](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.13/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.13/x64/lib
ImportError while loading conftest '/home/runner/work/conbench/conbench/legacy/tests/conftest.py'.
legacy/tests/conftest.py:1: in <module>
    import flask
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/flask/__init__.py:[5](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:5): in <module>
    from .app import Flask as Flask
/opt/hostedtoolcache/Python/3.10.13/x[6](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:6)4/lib/python3.[10](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:10)/site-packages/flask/app.py:30: in <module>
    from werkzeug.urls import url_quote
E   ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/opt/hostedtoolcache/Python/3.10.[13](https://github.com/conbench/conbench/actions/runs/6487345281/job/17617445534?pr=1496#step:10:14)/x64/lib/python3.10/site-packages/werkzeug/urls.py)
Error: Process completed with exit code 4.
```

Probably this is a deprecation/versioning error. I believe this package is only used in testing.

--- a/legacy/requirements-dev.txt
+++ b/legacy/requirements-dev.txt
@@ -1,3 +1,5 @@
 conbench[server]
 Flask==2.2.3
 pytest>=7.0.0
+# Flask 2.2.3 isn't compatible with werkzeug >= 3
+werkzeug<3

